### DISCUSSION TITLE
[New Feature]: Made it easier for developers to contribute to winget.

### DISCUSSION THREAD

**Discussion by gnpaone** [microsoft/winget-pkgs#56758] [Show and tell]
### Description of the new feature/enhancement

I have made a Github Action currently in the [Github Marketplace](https://github.com/marketplace/actions/winget-push-test) which makes it easier for the users to push/submit their packages to this repository as well as made it a lot easier to test the pushing of the package via the workflow and experience how it's going to look like (this is the place where I struggled at the beginning when I want to test my package long ago - so this solution). (just my little contribution to the developers). An example usage is here: https://github.com/gnpaone/MSEdgeRedirect/blob/main/.github/workflows/winget-test.yml
It helps users/developers a lot. Any reviews or suggestions is greatly appreciated.

### Proposed technical implementation details (optional)

As I mentioned in here (https://github.com/gnpaone/winget-push-test/blob/master/README.md) default submission will be done to this repository or can try submitting to test repositories before directly submitting here, all done using Github workflow. One example is here: https://github.com/gnpaone/MSEdgeRedirect/blob/main/.github/workflows/winget-test.yml
I did tests to verify the working.
```yaml
uses: gnpaone/winget-push-test@v1
with:
  url32:
  url64:
  version:
  token:
  packageid:
  user:
  repo:
```

**Comment by Trenly** [comment]
Not sure this is an "Issue", but it's more of a discussion topic

**Comment by jedieaston** [comment]
+1 on the discussion topic (you should be able to convert it in the sidebar), but this looks really cool. Thanks!

How long has GitHub let you package Actions that run on Windows only? That was the major reasoning behind https://github.com/microsoft/winget-create/discussions/160

**Comment by gnpaone** [comment]
Yah, it's kinda discussion topic. I think this repo owners only can convert this into discussion https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion

> 
> How long has GitHub let you package Actions that run on Windows only? That was the major reasoning behind [microsoft/winget-create#160](https://github.com/microsoft/winget-create/discussions/160)

The user mentions specifically to run in Windows in Github workflow so those Actions run in Windows only right. Commands like `runs-on: windows-latest` and `shell: powershell`

**Comment by gnpaone** [comment]
Actually, it is possible to specifically mentioning to run wingetcreate update in linux or macos via Github workflow but runtime will increase by 1-2 mins

**Comment by vedantmgoyal9** [comment]
I found another one: https://github.com/isaacrlevin/winget-publish-action

**Reply by gnpaone** [reply]
Nice 😊, but there are few features missing in this that's in mine 😅, testing the release to for example forked winget-pkg repo via github-actions kinda like https://github.com/microsoft/winget-create#testing-the-client 😊

**Reply by gnpaone** [reply]
I will include appx too in mine soon in my next version 😉, any more new features/additions in winget-pkgs @vedantmgoyal2009?

**Reply by vedantmgoyal9** [reply]
@gnpaone well, your action uses wingetcreate so I don't understand what additional changes you've to make to support appx?  wingetcreate already detects metadata from every type of Installer (including appx/msix).

**Reply by gnpaone** [reply]
https://github.com/microsoft/winget-create#using-the-msixbundle I mean like this? 🙃

**Reply by vedantmgoyal9** [reply]
That isn't required in my opinion, wingetcreate msix and wingetcreate exe are same.

**Reply by gnpaone** [reply]
> That isn't required in my opinion, wingetcreate msix and wingetcreate exe are same.

Ohh ok, understood

**Comment by gnpaone** [comment]
Also recently read in an article that winget supports portable apps too https://winaero.com/winget-now-supports-portable-apps/, so any additions in manifests parameters/arguments recently (I'm asking generally, not only for portable apps)?

**Reply by gnpaone** [reply]
And there's some discussion going on zip file installer integration, any progress on that?

**Comment by gnpaone** [comment]
Hi @vedantmgoyal2009, I've seen winget releaser 😏. A small doubt, is the traditional way of submitting obsolete or something?

**Reply by vedantmgoyal9** [reply]
@gnpaone I don't understand what you mean to say.

**Reply by gnpaone** [reply]
I mean submitting directly to winget-pkg as before instead of like winget releaser is still supported right?

**Reply by vedantmgoyal9** [reply]
Yes, the action basically simplifies the process of getting versions, urls, updating manifests and submitting pull requests into one step so that everyone doesn't need to write their own custom workflow that does the same process. In other words, you can say it like a _script_, which can be customized according to one's needs and used.

**Reply by gnpaone** [reply]
Ok
